### PR TITLE
feat: add create_gtfs_feed endpoint to operations API

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**
Part of #1412 

This PR introduces the POST endpoint to the GTFS feeds. The GTFS-RT feeds POST endpoint will be a follow-up PR. The feeds created using the operational POST endpoints will have IDs with the prefix `md-` and a sequential integer, e.g., `md-1`.

## From our AI friend

This pull request adds support for converting dictionary representations of ORM models to actual SQLAlchemy ORM objects for several entities, including feeds, external IDs, related links, locations, and redirects. It introduces utility functions and new methods for each model to handle this conversion, with careful handling of nested relationships and edge cases. Additionally, comprehensive unit tests are added to verify the new functionality, including correct mapping, sorting, and error handling.

**ORM Conversion Methods and Utilities**

* Added `to_orm_from_dict` methods to `FeedImpl`, `ExternalIdImpl`, `FeedRelatedLinkImpl`, `LocationImpl`, and `RedirectImpl` to convert dictionaries into their respective ORM objects, supporting nested collections and sorting where needed. [[1]](diffhunk://#diff-72eb0b08d47365bbe662ca1147b835ea12c6a104094852526635f3737e066686R34-R78) [[2]](diffhunk://#diff-d72ea44f68443656e8eed6007ecc27851b6de53519195e786a04e315d9084577R24-R32) [[3]](diffhunk://#diff-f9fe0d37ab1b6509e603e1270a0fc5d245458748365d283294b554cd1b7a2d06R24-R35) [[4]](diffhunk://#diff-d36258c9a57bec7cf5a2eb51bf1fb4806b9c39c0cc0aba7fe7e62fed1eb57605R35-R46) [[5]](diffhunk://#diff-226cef9880941519957c22c93afa8d4249bc4cb576a5367058cab41d29c53127R27-R41)
* Introduced the utility function `create_or_get_location` in `locations_utils.py` to handle location creation and retrieval, including ISO country code lookup and ID generation.

**Testing and Validation**

* Added and expanded unit tests for the new conversion methods in `FeedImpl`, `ExternalIdImpl`, `FeedRelatedLinkImpl`, and `LocationImpl`, ensuring correct conversion, sorting, handling of empty inputs, and error cases. [[1]](diffhunk://#diff-7279aa98d00d832f839b614d544c7404fd03114d7ffb3858093e64d633d8c7ceR163-R256) [[2]](diffhunk://#diff-8d2113cbdfa7a0c4cfa9709cc9df373aa6a983ab8eb92bcbbf7c937cb79d5e76L39-R59) [[3]](diffhunk://#diff-a0a137b1e5186004eb8ab7afb3030161e2f017392f9fc340c5d3d16bb836007dR1-R62) [[4]](diffhunk://#diff-9d799d350e1cfd25bce043a6bdc6738a1586354372eef83f125b6778fc88541cR20-R84)
* Used mocking in tests to isolate and validate location and redirect creation logic, including error handling for missing keys and database dependencies. [[1]](diffhunk://#diff-9d799d350e1cfd25bce043a6bdc6738a1586354372eef83f125b6778fc88541cR20-R84) [[2]](diffhunk://#diff-eaf26560a336d0945d1adc1a9d603926994eac8bc5214b595e1fd2760e4d060eR2)

**Imports and Refactoring**

* Updated imports in several files to support new conversion logic and utility functions, including database session management and model references. [[1]](diffhunk://#diff-72eb0b08d47365bbe662ca1147b835ea12c6a104094852526635f3737e066686R4-R9) [[2]](diffhunk://#diff-d36258c9a57bec7cf5a2eb51bf1fb4806b9c39c0cc0aba7fe7e62fed1eb57605R1-R7) [[3]](diffhunk://#diff-226cef9880941519957c22c93afa8d4249bc4cb576a5367058cab41d29c53127L1-R4) [[4]](diffhunk://#diff-9d799d350e1cfd25bce043a6bdc6738a1586354372eef83f125b6778fc88541cR2) [[5]](diffhunk://#diff-eaf26560a336d0945d1adc1a9d603926994eac8bc5214b595e1fd2760e4d060eR2)

**Minor Improvements**

* Improved test coverage for edge cases such as handling empty or None inputs, extra dictionary keys, and explicit empty collections. [[1]](diffhunk://#diff-7279aa98d00d832f839b614d544c7404fd03114d7ffb3858093e64d633d8c7ceR163-R256) [[2]](diffhunk://#diff-8d2113cbdfa7a0c4cfa9709cc9df373aa6a983ab8eb92bcbbf7c937cb79d5e76L39-R59) [[3]](diffhunk://#diff-a0a137b1e5186004eb8ab7afb3030161e2f017392f9fc340c5d3d16bb836007dR1-R62)

These changes collectively enable robust conversion from dictionary payloads to ORM objects, supporting more flexible data ingestion and manipulation workflows.

**Expected behavior:** 

- A GTFS is created when the  [POST]v1/operations/feeds/gtfs endpoint is created.
- The endpoint returns an error when a is called with a feed that matches any URL already in the system that is not marked as deprecated

**Testing tips:**
Local testing
- Update your local DB
- Start the operations API
- Hit the endpoint with the following request
```
curl --request POST \
  --url http://localhost:8080/v1/operations/feeds/gtfs \
  --header 'content-type: application/json' \
  --data '{
  "status": "active",
  "external_ids": [
    {
      "external_id": "1210  ",
      "source": "mdb "
    }
  ],
  "provider": "Los Angeles Department of Transportation (LADOT, DASH, Commuter Express)    ",
  "feed_name": "Bus",
  "note": "string",
  "feed_contact_email": "someEmail@ladotbus.com ",
  "source_info": {
    "producer_url": "https://ladotbus.com/gtfs10",
    "authentication_type": 2,
    "authentication_info_url": "https://apidevelopers.ladottransit.com",
    "api_key_parameter_name": "Ocp-Apim-Subscription-Key",
    "license_url": "https://www.ladottransit.com/dla.html"
  },
  "redirects": [
    {
      "target_id": "mdb-10",
      "comment": "Redirected because of a change of URL."
    }
  ],
  "operational_status": "published",
  "official": true,
  "related_links": [
    {
      "code": "next_1",
      "description": "The next",
      "url": "https://the.next.org"
    }
  ],
  "locations": [
    {
      "country_code": "US",
      "country": "United States",
      "subdivision_name": "California",
      "municipality": "Angeles"
    }
  ]
}'
```
- Expect a response with the newly created information
- If the operation is repeated with the same request, the creation of the feed is rejected as the producer's URL matches

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: functions-python/operations_api/src/feeds_operations/impl/feeds_operations_impl.py [hunk 1]
```diff
@@ -292,3 +302,50 @@ async def fetch_feed(data_type, session, update_request_feed):
...
+            new_feed.stable_id = f"md-{next_val}"
+        client_provided_id = bool(getattr(new_feed, "id", None))
+        if not client_provided_id:
+            new_feed.id = new_feed.stable_id
+        db_session.add(new_feed)
+        db_session.commit()
+        created_feed = db_session.get(Gtfsfeed, new_feed.id)
+        logging.info("Created new GTFS feed with ID: %s", new_feed.stable_id)
+        payload = OperationGtfsFeedImpl.from_orm(created_feed).model_dump()
+        return JSONResponse(status_code=201, content=jsonable_encoder(payload))
```
Comment (cka-y) [inline functions-python/operations_api/src/feeds_operations/impl/feeds_operations_impl.py]: [question] after successful creation of a feed shouldn't we trigger a dataset download? (maybe out of scope for this issue)

Comment (davidgamez) [reply functions-python/operations_api/src/feeds_operations/impl/feeds_operations_impl.py]: This is a good point, let's address it in the same issue in a follow-up PR.
